### PR TITLE
Update CircleCI config

### PR DESCRIPTION
**Pull request by osanoj**
# Why?
It was a little bit old.

# What?
- Use latest Xcode version.
- See if we can use the M1 runner instead of Intel.

# Version Change
No changes to the library

### CODE REVIEW

**Review by robinsalehjan — APPROVED** [review]

**Comment by osanoj** [comment]
Screenshot 2023-12-05 at 11 24 51

_"See if we can use the M1 runner instead of Intel."_ -> Nope

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,7 +2,8 @@ version: 2.1
 jobs:
   build-and-test:
     macos:
-      xcode: 15.0.0
+      xcode: 15.0.1
```
Comment (BjornRuud) [inline .circleci/config.yml:5]: Can this be specified less specific? Having to update for each patch version seems unnecessary. What about the resource_class, can that be more general?

**Review by BjornRuud — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (osanoj) [reply .circleci/config.yml:5]: Was trying to check in their docs, but found no example of how to be less specific, but could of course just try to say 15 and see what happens 🤷

Comment (AndreyMomot) [reply .circleci/config.yml]: It works for Travis to use just 15 or 15.0, we should try same here...

Comment (robinsalehjan) [reply .circleci/config.yml]: > It works for Travis to use just 15 or 15.0, we should try same here...

They are two completely different services though..

> Was trying to check in their docs, but found no example of how to be less specific, but could of course just try to say 15 and see what happens 🤷

They specify `15.0.0` in their docs https://circleci.com/docs/testing-ios/  but might just default to `15`

Comment (AndreyMomot) [reply .circleci/config.yml]: For sure

**Pull request closed without merging**